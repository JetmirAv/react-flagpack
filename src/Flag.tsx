import * as React from 'react'
import './Flag.scss'

import { isoToCountryCode } from 'flag-pack-core'

interface Props {
  code: String,
  size?: String,
  hasDropshadow?: Boolean,
  hasBorder?: Boolean,
  gradient?: 'top-down' | 'real-circular' | 'real-linear',
  hasBorderRadius?: Boolean,
  customBorderRadius?: any,
  className?: String
}

class Flag extends React.PureComponent<Props, void> {
  public render() {
    const {
      gradient = '',
      size = 'l',
      hasBorder = false,
      hasDropshadow = false,
      hasBorderRadius = true,
      code = 'NL',
      customBorderRadius,
      className
    } = this.props

    return (
      <div
        style={{
          borderRadius: customBorderRadius
        }}
        className={
          `flag
          ${gradient}
          size-${size}
          ${hasBorder ? 'border' : ''}
          ${hasDropshadow ? 'dropshadow' : ''}
          ${hasBorderRadius ? 'border-radius' : ''}
          ${className}`.replace(/\s\s+/g, ' ').trim()
      }>
        <img src={`https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/${size.toLowerCase()}/${isoToCountryCode(code).toUpperCase()}.svg?sanitize=true`}/>
      </div>
    )
  }
}

export default Flag
