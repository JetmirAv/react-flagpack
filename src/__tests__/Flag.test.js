import React from 'react'
import renderer from 'react-test-renderer'

import Flag from '../Flag.tsx'

test('Should render a small dutch flag', () => {
  const component = renderer.create(
    <Flag
      code="NL"
      gradient="real-linear"
      size="m"
      dropshadow
    />,
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
