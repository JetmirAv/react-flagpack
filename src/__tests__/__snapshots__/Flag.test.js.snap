// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a small dutch flag 1`] = `
<div
  className="flag real-linear size-m dropshadow border-radius undefined"
  style={
    Object {
      "borderRadius": undefined,
    }
  }
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/NL.svg?sanitize=true"
  />
</div>
`;
